cmd/kubeadm/app/apis/kubeadm/v1alpha3/doc.go

```diff
@@ -19,127 +19,255 @@ limitations under the License.
-//	    serverCertSANs:
-//	    -  "ec2-10-100-0-1.compute-1.amazonaws.com"
-//	    peerCertSANs:
-//	    - "10.100.0.1"
-//	  external:
-//	    endpoints:
-//	    - "10.100.0.1:2379"
-//	    - "10.100.0.2:2379"
-//	    caFile: "/etcd/kubernetes/pki/etcd/etcd-ca.crt"
-//	    certFile: "/etcd/kubernetes/pki/etcd/etcd.crt"
-//	    certKey: "/etcd/kubernetes/pki/etcd/etcd.key"
-//	networking:
-//	  serviceSubnet: "10.96.0.0/12"
-//	  podSubnet: "10.100.0.1/24"
-//	  dnsDomain: "cluster.local"
-//	kubernetesVersion: "v1.12.0"
-//	controlPlaneEndpoint: "10.100.0.1:6443"
-//	apiServerExtraArgs:
-//	  authorization-mode: "Node,RBAC"
-//	controlManagerExtraArgs:
-//	  node-cidr-mask-size: 20
-//	schedulerExtraArgs:
-//	  address: "10.100.0.1"
-//	apiServerExtraVolumes:
-//	- name: "some-volume"
-//	  hostPath: "/etc/some-path"
-//	  mountPath: "/etc/some-pod-path"
-//	  writable: true
-//	  pathType: File
-//	controllerManagerExtraVolumes:
-//	- name: "some-volume"
-//	  hostPath: "/etc/some-path"
-//	  mountPath: "/etc/some-pod-path"
-//	  writable: true
-//	  pathType: File
-//	schedulerExtraVolumes:
-//	- name: "some-volume"
-//	  hostPath: "/etc/some-path"
-//	  mountPath: "/etc/some-pod-path"
-//	  writable: true
-//	  pathType: File
-//	apiServerCertSANs:
-//	- "10.100.1.1"
-//	- "ec2-10-100-0-1.compute-1.amazonaws.com"
-//	certificatesDir: "/etc/kubernetes/pki"
-//	imageRepository: "k8s.gcr.io"
-//	unifiedControlPlaneImage: "k8s.gcr.io/controlplane:v1.12.0"
-//	auditPolicy:
-//	  # https://kubernetes.io/docs/tasks/debug-application-cluster/audit/#audit-policy
-//	  path: "/var/log/audit/audit.json"
-//	  logDir: "/var/log/audit"
-//	  logMaxAge: 7 # in days
-//	featureGates:
-//	  selfhosting: false
-//	clusterName: "example-cluster"
-//
-// TODO: The BootstrapTokenString object should move out to either k8s.io/client-go or k8s.io/api in the future
-// (probably as part of Bootstrap Tokens going GA). It should not be staged under the kubeadm API as it is now.
+// Package v1alpha3 defines the v1alpha3 version of the kubeadm config file format, that is a big step
+// forward the objective of graduate kubeadm config to beta.
 //
+// One of the biggest changes introduced by this release is the re-design of how component config
+// can be provided to kubeadm; this will enable a improved stability of the kubeadm config while the efforts for
+// the implementation of component config across Kubernetes ecosystem continues.
+//
+// Another important change is the separation between cluster wide setting and runtime or node specific
+// settings, that is functional to the objective to introduce support for HA clusters in kubeadm.
+//
+// Migration from old kubeadm config versions
+//
+// Please convert your v1alpha2 configuration files to v1alpha3 using the kubeadm config migrate command of kubeadm v1.12.x
+// (conversion from older releases of kubeadm config files requires older release of kubeadm as well e.g. v1alpha1, v1.11.x).
```

Comment:
Perhaps this needs to be elaborated a bit, in that kubeadm v1.11 should be used to migrate v1alpha1 to v1alpha2 and then kubeadm v1.12 should be used to translate v1alpha2 to v1alpha3.